src/main/java/me/zeroeightsix/kami/module/modules/misc/HighwayTools.kt

```diff
@@ -0,0 +1,829 @@
+    var pathing = false
+    private var isSneaking = false
+    private var stuckDetector = 0
+    private lateinit var currentBlockPos: BlockPos
+    private lateinit var startingBlockPos: BlockPos
+
+    // stats
+    private var totalBlocksPlaced = 0
+    private var totalBlocksDestroyed = 0
+    private var totalBlocksDistance = 0
+
+    private enum class DebugMessages {
+        NONE, IMPORTANT, ALL
+    }
+
+    fun isDone(): Boolean {
+        return blockQueue.size == 0
+    }
+
+    override fun onEnable() {
+        if (mc.player == null) {
+            disable()
+            return
+        }
+        startingBlockPos = mcPlayerPosFloored(mc)
+        currentBlockPos = startingBlockPos
+        playerHotbarSlot = mc.player.inventory.currentItem
+        lastHotbarSlot = -1
+        buildDirectionSaved = getPlayerCardinal(mc)
+
+        if (baritoneMode.value) {
+            baritoneSettingAllowPlace = BaritoneAPI.getSettings().allowPlace.value
+            BaritoneAPI.getSettings().allowPlace.value = false
+            if (!goalRender.value) {
+                baritoneSettingRenderGoal = BaritoneAPI.getSettings().renderGoal.value
+                BaritoneAPI.getSettings().renderGoal.value = false
+            }
+        }
+
+        playerHotbarSlot = mc.player.inventory.currentItem
+
+        refreshData()
+        printEnable()
+    }
+
+    override fun onDisable() {
+        if (mc.player == null) return
+
+        // load initial player hand
+        if (lastHotbarSlot != playerHotbarSlot && playerHotbarSlot != -1) {
+            mc.player.inventory.currentItem = playerHotbarSlot
+        }
+        if (isSneaking) {
+            mc.player.connection.sendPacket(CPacketEntityAction(mc.player, CPacketEntityAction.Action.STOP_SNEAKING))
+            isSneaking = false
+        }
+        playerHotbarSlot = -1
+        lastHotbarSlot = -1
+
+
+        if (baritoneMode.value) {
+            BaritoneAPI.getSettings().allowPlace.value = baritoneSettingAllowPlace
+            if (!goalRender.value) BaritoneAPI.getSettings().renderGoal.value = baritoneSettingRenderGoal
+            val baritoneProcess = BaritoneAPI.getProvider().primaryBaritone.pathingControlManager.mostRecentInControl()
+            if (baritoneProcess.isPresent && baritoneProcess.get() == HighwayToolsProcess) {
+                baritoneProcess.get().onLostControl()
+            }
+        }
+        printDisable()
+        totalBlocksPlaced = 0
+        totalBlocksDestroyed = 0
+        totalBlocksDistance = 0
+    }
+
+    override fun onUpdate() {
+        if (mc.playerController == null) return
+
+        if (baritoneMode.value) {
+            pathing = BaritoneAPI.getProvider().primaryBaritone.pathingBehavior.isPathing
+            if (!isDone()) {
+                if (!doTask()) {
+                    if (!pathing && !ModuleManager.getModuleT(LagNotifier::class.java)!!.paused && !ModuleManager.getModuleT(AutoObsidian::class.java)!!.active) {
```

Comment:
You don't need `ModuleManager.getModuleT() ` to get the instance now. You can just access it directly.
For example you can do `LagNotifier.paused`

Reply:
Thats cool